proxy/fadein.go

```diff
@@ -26,18 +26,18 @@ func fadeIn(lifetime time.Duration, duration time.Duration, exponent float64) fl
 	return math.Pow(float64(lifetime)/float64(duration), exponent)
 }
 
-func filterFadeIn(endpoints []routing.LBEndpoint, rt *routing.Route, registry *routing.EndpointRegistry, rnd *rand.Rand) []routing.LBEndpoint {
-	if returnFast(rt) {
+func (f *fadeIn) filterFadeIn(endpoints []routing.LBEndpoint, rt *routing.Route) []routing.LBEndpoint {
+	if f.noFadeIn(rt) {
 		return endpoints
 	}
 
 	now := time.Now()
-	threshold := rnd.Float64()
+	threshold := f.rnd.Float64()
 
 	filtered := make([]routing.LBEndpoint, 0, len(endpoints))
 	for _, e := range endpoints {
-		f := fadeIn(
-			now.Sub(registry.GetMetrics(e.Host).DetectedTime()),
+		f := f.fadeInScore(
+			now.Sub(f.endpointRegistry.GetMetrics(e.Host).DetectedTime()),
```

Comment:
Maybe extract this to a locak variable `age := now.Sub(f.endpointRegistry.GetMetrics(e.Host).DetectedTime())`